src/engine/routing_algorithms/alternative_path_mld.cpp

```diff
@@ -400,27 +471,35 @@ template <typename RandIt> RandIt filterUnpackedPathsBySharing(RandIt first, Ran
         return last;
 
     std::unordered_set<EdgeID> edges;
-    edges.reserve(size * shortest_path.edges.size() * (1. + kAtMostLongerBy));
-
-    edges.insert(begin(shortest_path.edges), begin(shortest_path.edges));
+    edges.reserve(size * shortest_path.edges.size() * (1.25));
 
-    const auto over_sharing_limit = [&](const auto &unpacked) {
+    edges.insert(begin(shortest_path.edges), end(shortest_path.edges));
 
+    const auto over_sharing_limit = [&](auto &unpacked) {
         if (unpacked.edges.empty())
         { // don't remove routes with single-node (empty) path
             return false;
         }
 
-        const auto not_seen = [&](const EdgeID edge) { return edges.count(edge) < 1; };
-        const auto different = std::count_if(begin(unpacked.edges), end(unpacked.edges), not_seen);
+        EdgeWeight total_duration = 0;
+        const auto add_if_seen = [&](const EdgeWeight duration, const EdgeID edge) {
+            auto edge_duration = facade.GetEdgeData(edge).duration;
+            total_duration += edge_duration;
+            if (edges.count(edge) > 0)
+            {
+                return duration + edge_duration;
+            }
+            return duration;
+        };
 
-        const auto difference = different / static_cast<double>(unpacked.edges.size());
-        BOOST_ASSERT(difference >= 0.);
-        BOOST_ASSERT(difference <= 1.);
+        const auto shared_weight =
+            std::accumulate(begin(unpacked.edges), end(unpacked.edges), EdgeWeight{0}, add_if_seen);
 
-        const auto sharing = 1. - difference;
+        unpacked.sharing = shared_weight / static_cast<double>(total_duration);
+        BOOST_ASSERT(sharing >= 0.);
+        BOOST_ASSERT(sharing <= 1.);
```

Comment:
These assertions fail to compile on debug because we no longer have a `sharing` var in scope; need to be adapted.